packages/client/tests/memory/_utils/generateMemoryUsageReport.ts

```diff
@@ -0,0 +1,75 @@
+import fs from 'fs/promises'
+
+import { MemoryTestDir } from './MemoryTestDir'
+
+export async function generateMemoryUsageReport(testDir: MemoryTestDir, usageOverTime: number[]) {
+  await fs.writeFile(testDir.reportPath, getHtmlTemplate(testDir, usageOverTime))
+}
+
+function getHtmlTemplate(testDir: MemoryTestDir, usageOverTime: number[]) {
```

Comment:
I tried couple of ASCII charting libraries for terminal, but the once I tried either did not work that well with this dataset (for example, not trying to fit the graph into the available space) or haven't been updated for ages, so I opted for html report instead.